vendors/cypress/boards/CY8CKIT_064S0S2_4343W/ports/wifi/iot_wifi.c

```diff
@@ -489,14 +489,19 @@ WIFIReturnCode_t WIFI_NetworkGet(WIFINetworkProfile_t *pxNetworkProfile, uint16_
 WIFIReturnCode_t WIFI_NetworkDelete(uint16_t usIndex)
 {
     char wifi_profile_key[WIFI_PROFILE_KEY_SIZE];
+    cy_rslt_t lookupResult = MTB_KVSTORE_ITEM_NOT_FOUND_ERROR;
 
     configASSERT(usIndex < CY_TOTAL_WIFI_PROFILES);
 
     get_wifi_profile_key(usIndex + CY_FIRST_WIFI_PROFILE, wifi_profile_key);
 
     if (cy_rtos_get_mutex(&wifiMutex, wificonfigMAX_SEMAPHORE_WAIT_TIME_MS) == CY_RSLT_SUCCESS)
     {
-        if ((mtb_kvstore_read(&kvstore_obj, wifi_profile_key, NULL, NULL) != CY_RSLT_SUCCESS) ||
+        lookupResult = mtb_kvstore_read(&kvstore_obj, wifi_profile_key, NULL, NULL);
+
+        /* Reading from the kvstore may result in an item not found error if
+         * deleting a nonexisting network. */
+        if (((lookupResult != CY_RSLT_SUCCESS) && (lookupResult != MTB_KVSTORE_ITEM_NOT_FOUND_ERROR)) ||
```

Comment:
How about simplifying a bit to make it more readable?

```c
        /* Do not attempt to delete in case of item not found error which is
         * returned when trying to delete a non-existent profile. */
        if( ( lookupResult == CY_RSLT_SUCCESS ) ||
            ( lookupResult != MTB_KVSTORE_ITEM_NOT_FOUND_ERROR )
        {
            if( mtb_kvstore_delete(&kvstore_obj, wifi_profile_key) != CY_RSLT_SUCCESS )
            {
                cy_rtos_set_mutex(&wifiMutex);
                return eWiFiFailure;
            }
        }
```

Reply:
Ok